react-front-end/tsrc/search/SearchPage.tsx

```diff
@@ -360,33 +360,51 @@ const SearchPage = ({ updateTemplate, advancedSearchId }: SearchPageProps) => {
       );
 
       setSearchPageOptions(state.options);
-      Promise.all([doSearch(state.options), getClassifications(state.options)])
-        .then(
-          ([result, classifications]: [
-            SearchPageSearchResult,
-            Classification[]
-          ]) => {
-            dispatch({
-              type: "search-complete",
-              result: { ...result },
-              classifications: [...classifications],
-            });
-            // Update history
-            history.replace({
-              ...history.location,
-              state: { searchPageOptions: state.options, filterExpansion },
-            });
-            // Save the value of wildcard mode to LocalStorage.
-            writeRawModeToStorage(state.options.rawMode);
-            // scroll back up to the top of the page
-            if (state.scrollToTop) window.scrollTo(0, 0);
-            // Allow downloading new search result.
-            setAlreadyDownloaded(false);
-          }
-        )
-        .catch(searchPageErrorHandler);
+      (async () => {
+        try {
+          const searchResult: SearchPageSearchResult = await doSearch(
+            state.options
+          );
+          // Do not list classifications in Advanced search mode.
+          const classifications: Classification[] =
+            advancedSearchId === undefined
```

Comment:
Based on the comments below, I wonder if for consistency this should be:

```suggestion
            !!advancedSearchId
```

But up to you, as I also don't really mind this way. I'm just mentioning it.

Reply:
I think here should be single `!`.